src/main/java/org/jenkinsci/plugins/ewm/steps/model/ExternalWorkspace.java

```diff
@@ -2,27 +2,35 @@
 
 import javax.annotation.Nonnull;
 import java.io.Serializable;
+import java.util.UUID;
 
 /**
  * POJO used to pass fields from one step to another.
  *
  * @author Alexandru Somai
  */
-public final class ExternalWorkspace implements Serializable {
+public class ExternalWorkspace implements Serializable {
 
     private static final long serialVersionUID = 1L;
 
+    private final UUID id;
     private final String diskPoolId;
     private final String diskId;
     private final String pathOnDisk;
 
     public ExternalWorkspace(@Nonnull String diskPoolId, @Nonnull String diskId, @Nonnull String pathOnDisk) {
+        this.id = UUID.randomUUID();
```

Comment:
I don't think I can use https://github.com/jenkinsci/unique-id-plugin for generating workspace unique id.
From my understanding, if I use `IdStore.makeId(Object o)` it stores a generated id into a file. But I don't think I need that kind of functionality.